driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -309,6 +309,19 @@ public static Bson facet(final Facet... facets) {
         return new FacetStage(asList(facets));
     }
 
+    /**
+     * Creates a function pipeline stage
+     *
+     * @param function  the function to use
+     * @return the new pipeline stage
+     * @mongodb.driver.manual reference/operator/aggregation/function/ $function
+     * @mongodb.server.release 4.4
+     * @since 4.1
+     */
+    public static Bson function(final Function function) {
```

Comment:
Why opt for a `Function` object over allowing the user set them directly via the API?  eg: `function(String body, List<string> args, String lang)`

Reply:
Is there precedent one way or the other in the builder API?

Reply:
`lookup` uses named args for the method over creating a class to represent a lookup.  Whereas complex API items like `facet` take a list of `Facet` Objects.

Reply:
and `unwind` takes values and then has an options class for optionals